phac-nml/irida-next
# Workflow executions: Handle metadata and attachments

## What does this PR do and why?
_Describe in detail what your merge request does and why._
Fixes #495 

Dependant on #501

During the CompletionService, add the following functionality
* attach outputs put onto SamplesWorkflowExecutions to the related Samples
  * these outputs use the same blob, so no file duplication
* merge output metadata to Sample objects
* Add step to flatten complex metadata arrays and hashes using dot notation

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Connect to a gh4gh_wes instance with related blob storage.
2. Run an workflow with some samples on it
3. Verify that after the completion step is finished the samples have attached output files
4. Verify that the attached output files are the same blob's as the blobs on the related SamplesWorkflowExecutions
5. Verify that the samples metadata has been updated

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
